drivers/entropy/entropy_stm32.c

```diff
@@ -487,6 +487,18 @@ static int entropy_stm32_rng_init(const struct device *dev)
 		(clock_control_subsys_t *)&dev_cfg->pclken);
 	__ASSERT_NO_MSG(res == 0);
 
+
+#if DT_INST_NODE_HAS_PROP(0, htcfg)
+#if DT_INST_NODE_HAS_PROP(0, htcfg1)
+	/* Write Magic number before writing configuration
+	 * Not all stm32 series have a Magic number
+	 */
+	LL_RNG_SetHealthConfig(dev_data->rng, DT_INST_PROP(0, htcfg1));
+#endif
+	/* Write RNG HTCR configuration */
+	LL_RNG_SetHealthConfig(dev_data->rng, DT_INST_PROP(0, htcfg));
+#endif
+
```

Comment:
Shouldn't it be done when the SoC resumes from a sleep state ?

Reply:
Nothing special to RNG. Like other peripherals, registers content is lost when wakeup from Standby/shutdown but in such case there is a full zephyr initialization is performed.